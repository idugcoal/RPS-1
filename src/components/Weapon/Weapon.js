import React from 'react';
import PropTypes from 'prop-types';
import style from './Weapon.css';

const Weapon = ({ name, onClick }) => (
  <div onClick={onClick}>
    <img className={style.weapon} src={`/static/images/${name}.png`} alt={name} />
  </div>
);

Weapon.propTypes = {
  name: PropTypes.string,
  onClick: PropTypes.func,
};

export default Weapon;
