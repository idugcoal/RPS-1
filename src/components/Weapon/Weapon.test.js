import React from 'react';
import { shallow } from 'enzyme';
import Weapon from './Weapon';

test('should render correctly', () => {
  const click = jest.fn();
  const wrapper = shallow(<Weapon
    name='rock'
    onClick={click}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should call onClick', () => {
  const click = jest.fn();
  const wrapper = shallow(<Weapon
    name='rock'
    onClick={click}
  />);
  wrapper.find('div').simulate('click');
  expect(click.mock.calls.length).toBe(1);
});
