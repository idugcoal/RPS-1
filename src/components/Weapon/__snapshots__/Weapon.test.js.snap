// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  onClick={[MockFunction]}
>
  <img
    alt="rock"
    src="/static/images/rock.png"
  />
</div>
`;
